renku/core/management/repository.py

```diff
@@ -148,6 +149,11 @@ def __attrs_post_init__(self):
             except subprocess.CalledProcessError:
                 pass
 
+    @property
+    def latest_version(self):
+        """Returns latest `SoftwareAgent` used in the repository."""
+        return SoftwareAgent.from_commit(self.repo.head.commit)
```

Comment:
This doesn't work if the newest commit was a commit done by the user (not renku), e.g.

```
>>> from renku.core.commands.migrate import migrations_check, migrations_versions
>>> migrations_versions()
('0.10.5.dev52', 'Panaetius')
```

Reply:
Yea, this PR was supposed to be draft. There is also one more issue present on the service when the source and destination are the same. I'll address this asap.